src/ralph/backends/data/base.py

```diff
@@ -169,28 +181,22 @@ def validate_backend_query(
             logger.error(msg, query_name, errors)
             raise BackendParameterException(msg % (query_name, errors)) from error
 
-    if isinstance(query, str):
-        try:
-            return query_class(query_string=query)
-        except ValidationError as error:
-            msg = "Invalid %s query string: %s"
-            errors = error.errors()
-            logger.error(msg, query_name, errors)
-            raise BackendParameterException(msg % (query_name, errors)) from error
+    if isinstance(query, query_class):
+        return query
+
+    if issubclass(query_class, BaseQuery) and isinstance(query, str):
```

Comment:
Why do we need `issubclass(query_class, BaseQuery) ` ? Shouldnt this always be True ?

Reply:
Good question) This is the key change of this PR - the `query_class` type is extended to `Union[BaseQuery, str, dict]`, which adds the need for this check

Reply:
Update: This change was removed as we chose to drop support for converting string queries to `BaseQuery` if the `query_class` is `BaseQuery`.